src/java/boa/functions/BoaIntrinsics.java

```diff
@@ -422,6 +422,18 @@ public static boolean iskind(final String s, final ChangedFile.FileKind kind) {
 			return null;
 		return s.peek();
 	}
+	
+	public static <T> T queue_poll(final java.util.Queue<T> q) {
```

Comment:
im curious why these are needed?

https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html#poll()

This indicates `null` is returned if empty.  So simply calling poll() on the queue is sufficient and we dont need helper function here?  Similar with peekFirst() (which we could use instead of peek()):

https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html#peekFirst()